
We build dietlibc with WANT_DYNAMIC so we get ctor/dtor handling, even
for static programs, as normally needed.

The patch zeros register 6 passed for dl_init to _dyn_start since otherwise
we end up in an inf. loop calling the _dyn_start (where r6 points to) again
and again ... This is NOT intended for upstream merge - just a quick hack
to get it working.

  - Rene Rebe <rene@exactcode.de>

diff -u dietlibc-0.28/ppc/start.S dietlibc-0.28-fixed/ppc/start.S
--- dietlibc-0.28/ppc/start.S	2004-09-11 14:37:57.000000000 +0200
+++ dietlibc-0.28-fixed/ppc/start.S	2005-05-03 23:47:09.000000000 +0200
@@ -27,7 +29,7 @@
 	stw	5,environ@l(14)
 
 #ifdef WANT_DYNAMIC
-	mr	6,7
+	xor	6,6,6
 	bl	_dyn_start
 #else
 #ifdef WANT_STACKGAP
